src/main/java/cz/tstrecha/timetracker/service/impl/UserServiceImpl.java

```diff
@@ -89,21 +85,57 @@ public UserDTO createUser(UserRegistrationRequestDTO registrationRequest, UserRo
         return userMapper.toDTO(savedUser);
     }
 
-    public RelationshipDTO createRelationship(RelationshipCreateUpdateRequestDTO request){
+    public RelationshipDTO createRelationship(RelationshipCreateUpdateRequestDTO request) {
```

Comment:
Always @Transactional if there is any interaction with method.

But careful - transaction using @Transactional is craeted only when this method is called from different bean. Transaction also ends (commits or rollbacks) only when the code's stacktrace reaches end of the whole bean, not just the method.